### DISCUSSION TITLE
Fetch async config before strapi establishes a connection to the db

### DISCUSSION THREAD

**Discussion by psmyrdek** [strapi/strapi#7818] [Legacy Topics]
We're using an external config server (like Consul) to store properties for different envs. I'd like to fetch those before strapi establishes a connection to the db. I was trying to play with database.js and server.js but the order isn't correct. What would be the right place to fetch my config then? A hook?

**Comment by derrickmehaffy** [comment]
In this case you may want to use a server.js file to fetch the information before you start Strapi. You can use something like:

```js
const strapi = require('strapi');

// Your custom code to fetch information from Consul

strapi(/* {...} */).start();
```

**Reply by psmyrdek** [reply]
Oh right! Thanks! Qq - what can I put inside of `strapi(/* {...} */)` - is it documented somewhere?

**Reply by derrickmehaffy** [reply]
Tbh I'm not really sure and we don't have that documented as we don't generally recommend it's usage (it's not needed except in cases like yours)
